### PR TITLE
EarlGrey Source Sync 

### PR DESCRIPTION
**Pull request by phsong-google**
EarlGrey Source Sync

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,13 +11,9 @@ Baseline: [0cdda9c]
...
 
 ### Bug Fixes
-* Updated the EarlGrey API for Swift 3.0 as per the latest guidelines.
-* Improved web tests to work with current google.com UI.
-* Fixed a bug in the visibility checker for 32bit platform.
-* Fixed flakiness caused by NSDate issues in EarlGreyExampleSwiftTests.
-
-### Enhancements
-* Added a travis hook to stop CI runs for docs-only changes.
+* Fixed EarlGrey.swift for Swift 3.0
```
Comment (khandpur) [inline CHANGELOG.md]: this needs to be reverted. The changes before are more thorough.

File: EarlGrey/Core/GREYElementFinder.m [hunk 2]
```diff
@@ -37,8 +37,8 @@ - (NSArray *)elementsMatchedInProvider:(id<GREYProvider>)elementProvider {
   NSParameterAssert(elementProvider);
   I_CHECK_MAIN_THREAD();
   NSMutableArray *matchingElements = [[NSMutableArray alloc] init];
-  for (id element in [elementProvider dataEnumerator]) {
-    @autoreleasepool {
+  @autoreleasepool {
```
Comment (khandpur) [inline EarlGrey/Core/GREYElementFinder.m]: revert. please do a thorough check first. sync github to internal then internal to external please.

**Comment by khandpur** [comment]
Can you add the bugs this PR is addressing?

**Comment by khandpur** [comment]
Should we drop this?

**Comment by tirodkar** [comment]
Closing this PR since it's been replicated in smaller ones.

**Pull request closed without merging**